src/Engine/Renderer/RenderTechnique/RenderParameters.inl

```diff
@@ -17,7 +17,9 @@ inline void RenderParameters::TParameter<T>::bind( const ShaderProgram* shader )
 }
 
 inline void RenderParameters::TextureParameter::bind( const ShaderProgram* shader ) const {
-    shader->setUniform( m_name, m_texture, m_texUnit );
+    if ( m_texUnit == -1 ) { shader->setUniformTexture( m_name, m_texture ); }
+    else
+    { shader->setUniform( m_name, m_texture, m_texUnit ); }
```

Comment:
It is never explained, neither in the documentation nor in the code, that the value of `m_texUnit` could change the behavior of this function. Could you somehow document this ?

Reply:
class documentation now explain the difference.